pkg/cluster/internal/create/actions/config/config.go

```diff
@@ -185,8 +185,8 @@ func (a *Action) Execute(ctx *actions.ActionContext) error {
 					return errors.Wrap(err, "failed to write patched containerd config")
 				}
 				// restart containerd now that we've re-configured it
-				// skip if the systemd (also the containerd) is not running
-				if err := node.Command("bash", "-c", `! systemctl is-system-running || systemctl restart containerd`).Run(); err != nil {
+				// skip if containerd is not running
+				if err := node.Command("bash", "-c", `! pgrep --exact containerd || systemctl restart containerd`).Run(); err != nil {
```

Comment:
`pgrep` returns `0` on a match and `1` on no match.

```
EXIT STATUS
0      One or more processes matched the criteria. For pkill and
              pidwait, one or more processes must also have been
              successfully signalled or waited for.
1      No processes matched or none of them could be signalled.
2      Syntax error in the command line.
3      Fatal error: out of memory etc.
```
https://man7.org/linux/man-pages/man1/pgrep.1.html

Reply:
containerd when running has a process tree rooted by `containerd`, the child processes are the shim which is `containerd-shim`. `pgrep --exact [pattern]` matches only an exact pattern match.

Reply:
Why not pkill -HUP then?

Reply:
This shell command we crafted exits success when successful in any case and only non zero when something doesn't work. We can correctly distinguish success.

Reply:
```suggestion
				if err := node.Command("bash", "-c", `pgrep --exact containerd || systemctl restart containerd`).Run(); err != nil {
```

Reply:
we want to restart containerd if the pgrep fails, right? we don't need to negate then

Reply:
```
$ bash -c '! pgrep --exact containerd || echo "notfound"'
490904
notfound
$ bash -c 'pgrep --exact containerd || echo "notfound"
490904
$ bash -c 'pgrep --exact container || echo "notfound"'
notfound
```

Reply:
If the pgrep fails there is no containerd process found, so what are we restarting? 🙃

Reply:
lol, what I was thinking ...